metricbeat/module/elasticsearch/cluster_stats/data_test.go

```diff
@@ -21,10 +21,12 @@ package cluster_stats
 
 import (
 	"testing"
-
-	"github.com/elastic/beats/v7/metricbeat/module/elasticsearch"
 )
 
 func TestMapper(t *testing.T) {
-	elasticsearch.TestMapperWithInfo(t, "./_meta/test/cluster_stats.*.json", eventMapping)
+	// TODO This test must be skipped until we find a reasonable way to test the event mapping when it has to do extra
+	// HTTP calls
```

Comment:
Could you mock an HTTP server using [`http.NewServeMux`](https://golang.org/pkg/net/http/#NewServeMux)? IIRC we are doing this in a few places in the Beats codebase if you want to try and follow some examples.

Reply:
Ah yes, the problem I had with this test is the `helper.HTTP` which is not easy to use in tests. Actually it's not used in any test file apart from some fake test in rabbitmq. The cleanest way to fix it is refactoring the helper as you can see in this branch https://github.com/elastic/beats/pull/22573 with the solution applied as example.

Reply:
Understood, the problem is that you need to pass in the Metricset and that's not easy to construct in tests. 

I took a look at the refactoring PR (#22573) and it LGTM.